lib/upipe-dveo/upipe_dveo_asi_sink.c

```diff
@@ -0,0 +1,765 @@
+    unsigned long int tmp;
+
+    memset (data, 0, sizeof (data));
+    if ((ret = util_read (name, data, sizeof (data))) < 0) {
+        return ret;
+    }
+    tmp = strtoul (data, &endptr, 0);
+    if (*endptr != '\n') {
+        return -1;
+    }
+    *val = tmp;
+    return ret;
+}
+
+/** @internal @This asks to open the given device.
+ *
+ * @param upipe description structure of the pipe
+ * @param path relative or absolute path of the file
+ * @return an error code
+ */
+static int upipe_dveo_asi_sink_open(struct upipe *upipe)
+{
+#define BYPASS_MODE 1
+
+    struct upipe_dveo_asi_sink *upipe_dveo_asi_sink = upipe_dveo_asi_sink_from_upipe(upipe);
+    char path[20], sys[50], buf[20];
+    memset(buf, 0, sizeof(buf));
+
+    static const char dev_fmt[] = "/dev/asitx%u";
+    static const char sys_fmt[] = "/sys/class/asi/asitx%u/%s";
+    static const char dvbm_sys_fmt[] = "/sys/class/dvbm/%u/%s";
+
+    snprintf(sys, sizeof(sys), sys_fmt, upipe_dveo_asi_sink->card_idx, "timestamps");
+    snprintf(buf, sizeof(buf), "%u\n", 2);
+    if (util_write(sys, buf, sizeof(buf)) < 0) {
+        upipe_err_va(upipe, "Couldn't set timestamp mode (%m)");
+        return UBASE_ERR_EXTERNAL;
+    }
+
+    snprintf(sys, sizeof(sys), sys_fmt, upipe_dveo_asi_sink->card_idx, "bufsize");
+    snprintf(buf, sizeof(buf), "%u\n", 6*(188+8)); /* minimum is 1024 */
+    if (util_write(sys, buf, sizeof(buf)) < 0) {
+        upipe_err_va(upipe, "Couldn't set buffer size (%m)");
+        return UBASE_ERR_EXTERNAL;
+    }
+
+    snprintf(sys, sizeof(sys), sys_fmt, upipe_dveo_asi_sink->card_idx, "buffers");
+    snprintf(buf, sizeof(buf), "%u\n", 500);
+    if (util_write(sys, buf, sizeof(buf)) < 0) {
+        upipe_err_va(upipe, "Couldn't set # of buffers (%m)");
+        return UBASE_ERR_EXTERNAL;
+    }
+
+    snprintf(path, sizeof(path), dev_fmt, upipe_dveo_asi_sink->card_idx);
+    int fd = open(path, O_WRONLY | O_NONBLOCK | O_CLOEXEC);
+    if (unlikely(fd < 0)) {
+        upipe_err_va(upipe, "can't open file %s (%m)", path);
+        return UBASE_ERR_EXTERNAL;
+    }
+
+    snprintf(sys, sizeof(sys), dvbm_sys_fmt, upipe_dveo_asi_sink->card_idx, "bypass_mode");
+    snprintf(buf, sizeof(buf), "%u", BYPASS_MODE);
+    util_write(sys, buf, sizeof(buf)); /* Not all cards have this so don't fail */
+
+    unsigned int cap;
+    if (ioctl(fd, ASI_IOC_TXGETCAP, &cap) < 0) {
+        upipe_err_va(upipe, "can't get tx caps (%m)");
+        goto error;
+    }
+
+    unsigned long int bufsize;
+    snprintf(sys, sizeof(sys), sys_fmt, upipe_dveo_asi_sink->card_idx, "bufsize");
+    if (util_strtoul(sys, &bufsize) < 0) {
+        upipe_err(upipe, "Couldn't read buffer size");
+        goto error;
+    }
+
+    unsigned long int mode;
+    snprintf(sys, sizeof(sys), sys_fmt, upipe_dveo_asi_sink->card_idx, "mode");
+    if (util_strtoul(sys, &mode) < 0) {
+        upipe_err(upipe, "Couldn't read buffer size");
+        goto error;
+    }
+
+    unsigned int packetsize;
```

Comment:
Dead code.

Reply:
Which code is dead? github web UI and email do not show the same code and I'm not sure what's wrong here.

Reply:
The packetsize calculation doesn't seem used anywhere.